### ISSUE TITLE
Firefox cleaner file

### ISSUE THREAD

**Issue by Hintsmth** [bleachbit/bleachbit#250]
Where can I find a Firefox cleaner file? I don't see it in the folder 'cleaners'.

**Comment by az0** [comment]
It's in Cleaner.py

See also https://bugs.launchpad.net/bleachbit/+bug/797995

**Comment by Hintsmth** [comment]
@az0 still don't get it. could you please provide me with information where I can find this?

**Comment by ROCKNROLLKID** [comment]
Cleaner.py is BleachBits built in cleaning rules that are separate from CleanerML (aka the ones in Cleaners folder). The Python cleaners will require some Python knowledge to edit them. The source file is here: https://github.com/bleachbit/bleachbit/blob/master/bleachbit/Cleaner.py

**Comment by Hintsmth** [comment]
@ROCKNROLLKID thank you but I don't really understand. I'm just a user not a programmer.

**Comment by Hintsmth** [comment]
@ROCKNROLLKID I've this reread and I got the idea. Is there any xml file for Firefox?

**Comment by ROCKNROLLKID** [comment]
It's OK. I am not a programmer either. We do want to move all the cleaners to CleanerML sometime in the future. I can't give you an exact release date on that, though, so you will have to be patient.

**Comment by Hintsmth** [comment]
@ROCKNROLLKID thank you! I'll be waiting. :)

**Comment by user139** [comment]
I posted this on the BleachBit Forums, but there was no response. So...

I found a zipped CleanerML file (forgot where) for Firefox - http://bit.ly/2Au3zZa
I tried to make it compatible with Pale Moon (though for Linux only) -  http://bit.ly/2knYjQV
Everything seems to work except for URL history.
In BB version 1.0 (which is available in the Ubuntu 14.04 repositories) the "URL history" item doesn't even show up in the menu. In 1.12 (as well as 1.17 beta) it does show up, but if ticked, throws an error:

> Error in execution of None
Traceback (most recent call last):
File "/usr/share/bleachbit/Worker.py", line 84, in execute
for ret in cmd.execute(self.really_delete):
AttributeError: 'NoneType' object has no attribute 'execute'

Any help with this? Also, is there any chance we could have proper compatibility with Firefox-based browsers before June 2018 (Firefox 52 ESR EOL)?
